core/src/main/java/com/linecorp/armeria/server/HttpStatusException.java

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright 2017 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package com.linecorp.armeria.server;
+
+import static java.util.Objects.requireNonNull;
+
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+
+import com.linecorp.armeria.common.Flags;
+import com.linecorp.armeria.common.HttpStatus;
+import com.linecorp.armeria.common.util.Exceptions;
+
+/**
+ * A {@link RuntimeException} that is raised when an armeria internal http exception has occurred.
```

Comment:
- IIUC, a user can also throw this exception, so it's not an internal exception. How about:

  ```
  A {@link RuntimeException} that is raised to send a simplistic HTTP response with minimal content by a {@link Service}.
  ```

- How about adding `@see HttpResponseException`?